cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java

```diff
@@ -137,14 +143,13 @@ public final String toString() {
 
   @Override
   public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
+    if (this == o) return true;
+    if (!(o instanceof EntityId)) return false;
+
     EntityId entityId = (EntityId) o;
-    return Objects.equals(entity, entityId.entity);
+
+    return getEntity() == entityId.getEntity();
```

Comment:
There is no `getEntityType` method in `EntityId`. `getEntity()` returns `EntityType`. Doesn't `return Objects.equals(this, entityId);` recursively call this same `equals` method? @poornachandra helps me to use intellij to generate this `equals` method automatically